rules/naming/src/Rector/Property/MakeBoolPropertyRespectIsHasWasMethodNamingRector.php

```diff
@@ -0,0 +1,200 @@
+use PhpParser\Node\VarLikeIdentifier;
+use PHPStan\Type\BooleanType;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\Naming\Tests\Rector\Property\MakeBoolPropertyRespectIsHasWasMethodNamingRector\MakeBoolPropertyRespectIsHasWasMethodNamingRectorTest
+ */
+final class MakeBoolPropertyRespectIsHasWasMethodNamingRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Renames property to respect is/has/was method naming', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    private $full = false;
+
+    public function isFull()
+    {
+        return $this->full;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    private $isFull = false;
+
+    public function isFull()
+    {
+        return $this->isFull;
+    }
+
+}
+CODE_SAMPLE
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $prefixedClassMethods = $this->getPrefixedClassMethods($node);
+        if ($prefixedClassMethods === []) {
+            return null;
+        }
+
+        $currentName = $this->getName($node);
+        $expectedName = $this->resolveExpectedName($prefixedClassMethods, $currentName);
+        if ($expectedName === null) {
+            return null;
+        }
+
+        if ($currentName === $expectedName) {
+            return null;
+        }
+
+        $this->renameProperty($node, $currentName, $expectedName);
+
+        return $node;
+    }
+
+    private function shouldSkip(Property $property): bool
+    {
+        if (! $property->isPrivate()) {
+            return true;
+        }
+        return ! $this->skipNonBooleanType($property);
+    }
+
+    /**
+     * @return ClassMethod[]
+     */
+    private function getPrefixedClassMethods(Property $property): array
+    {
+        $name = $this->getName($property);
+        if ($name === null) {
+            return [];
+        }
+
+        $classLike = $property->getAttribute(AttributeKey::CLASS_NODE);
+        if ($classLike === null) {
+            return [];
+        }
+
+        $classMethods = $this->betterNodeFinder->findInstanceOf($classLike, ClassMethod::class);
+        return array_filter($classMethods, function (ClassMethod $classMethod): bool {
+            $classMethodName = $this->getName($classMethod);
+            return Strings::match($classMethodName, '#^(is|was|has)[A-Z].+#') !== null;
```

Comment:
Maybe "can" too?

Reply:
I was also thinking about "are", 'were", "have" and 'had'. Maybe it is better to use some configuration param for it...

@TomasVotruba, what do you think?

Reply:
No need to configure it now, it would be too much detailed.
I think regex should be enough.